### PR TITLE
moving to new ABAC model

### PR DESCRIPTION
**Pull request by kris-watts-gravwell**
Hey Fritz

### CODE REVIEW

File: client/types/abac_test.go [hunk 1]
```diff
@@ -223,7 +208,9 @@ func TestAddRemove(t *testing.T) {
 	}
 
 	//change to default allow and recheck
```
Comment (ashnwade) [inline client/types/abac_test.go:210]: I think this comment is no longer correct. Always default deny

File: client/types/abac_test.go [hunk 2]
```diff
@@ -233,19 +220,13 @@ func TestAddRemove(t *testing.T) {
 	}
 
 	//set a few overrides (they should NOT be set)
```
Comment (ashnwade) [inline client/types/abac_test.go:222]: grants instead of overrides in comment

File: client/types/abac_test.go [hunk 3]
```diff
@@ -233,19 +220,13 @@ func TestAddRemove(t *testing.T) {
...
-	//switch default
-	cs.Default = DefaultDeny
-	if !cs.Has(Search) {
-		t.Fatal("capability is not allowed after override")
-	}
-
 	//remove override
-	if !cs.ClearOverride(Search) {
+	if !cs.Clear(Search) {
 		t.Fatal("failed to clear override")
```
Comment (ashnwade) [inline client/types/abac_test.go:230]: grants

File: client/types/abac_test.go [hunk 4]
```diff
@@ -281,78 +262,53 @@ func TestOverlap(t *testing.T) {
...
-	ud.Groups[1].ABAC.Capabilities.SetOverride(Search)
-	if ud.HasCapability(Search, globalDefaultAllow.Capabilities) {
-		t.Fatal("User has Search capability after group explicit denial")
+	//allow search
+	ud.Groups[1].ABAC.Capabilities.Set(Search)
+	if !ud.HasCapability(Search) {
+		t.Fatal("User has Search capability after group set")
 	}
 
 	//now swap the user assigned default and explicitely allow
```
Comment (ashnwade) [inline client/types/abac_test.go:278]: comment out of date

**Review by ashnwade — COMMENTED** [review]
Evan and I looked at this together on a meet. Functionally, looks good. Just a few comments and error messages to clean up.

**Comment by kris-watts-gravwell** [comment]
fixed up comments and messages, @ashnwade PTAL

**Review by ashnwade — APPROVED** [review]
LGTM

**Pull request merged**